rs/https_outcalls/consensus/src/gossip.rs

```diff
@@ -16,10 +16,10 @@ use ic_types::{
 };
 use std::{collections::BTreeSet, sync::Arc};
 
-/// The upper bound of how many shares of HTTP outcall requests unknown to the local replica will be
-/// requested via P2P in advance. Since the request ids are strictly increasing, we can simply add
-/// the constant to the latest known request id.
-const MAX_NUMBER_OF_REQUESTS_AHEAD: u64 = 50;
+// We are aiming for about 100 req/s for http outcalls. Assuming that the priority function gets
+// called about once every 3 seconds, we do not expect the number of requests to grow from one call
+// to another by about 100 http outcalls + 15 other management canister calls per second.
+const MAX_NUMBER_OF_REQUESTS_AHEAD: u64 = 3 * (100 + 15);
```

Comment:
Instead of hard-coding the factor `3`, could use the priority function poll delay directly? Or add a unit test to describe the relation between the two constants?

Reply:
@tthebst this would require me to make `PRIORITY_FUNCTION_UPDATE_INTERVAL` in `p2p/artifact_downloader/src/fetch_artifact/download.rs` public to be used either here or in a unit test. Or export this consts, but implement the unit test in your module.

Any particular preference?

Reply:
I would prefer to not depend `PRIORITY_FUNCTION_UPDATE_INTERVAL` and keeping as is with your reasoning.  This constant should anyway not be the bottleneck and if we set it high enough we are fine. 

If we really want to define constants that relate to each other we should have some constants crate to not pollute the dependent graph even more.

Reply:
@eichhorl since we're anyway approximating here, maybe we should just go with the change as is. It's unlikely that we ever reduce the invocation interval of the priority function. And if we increase it, we won't observe worse effects than if we do nothing but the usage of outcalls increases, right?

So bottom line is that nothing should break anyway.

Reply:
Fine with me 👍